apis/metal3.io/v1alpha1/hostfirmwaresettings_types.go

```diff
@@ -33,6 +33,16 @@ type SchemaReference struct {
 	Name string `json:"name"`
 }
 
+type SettingsConditionType string
+
+const (
+	// Indicates that the settings in the Spec are different than Status
+	ChangeRequested SettingsConditionType = "ChangeRequested"
```

Comment:
Well the comparison logic does happen in the provisioner during cleaning but we can avoid that whole step from the bmh if ChangeRequested is false. Its also a useful condition to present to the user, since updating the settings requires a reboot a user can determine if a reboot is required by looking at this condition

Reply:
Fine, maybe then a name like `UpdateRequested` could convey better the message?

Reply:
Sure, changed to use that.